### ISSUE TITLE
libunistring fails to build on UCRT

### ISSUE THREAD

**Issue by jeremyd2019** [msys2/MINGW-packages#8310]
After #8297 libunistring fails to build on UCRT with:
```
  x86_64-w64-mingw32-gcc -DHAVE_CONFIG_H -DNO_XMALLOC -I. -I../../libunistring-0.9.10/lib -I..  -I. -I../../libunistring-0.9.10/lib -I.. -I../../libunistring-0.9.10 -DIN_LIBUNISTRING -DDEPENDS_ON_LIBICONV=1 -D__USE_MINGW_ANSI_STDIO=1 -I/ucrt64/include  -march=x86-64 -mtune=generic -O2 -pipe -c ../../libunistring-0.9.10/lib/fseterr.c
  ../../libunistring-0.9.10/lib/fseterr.c: In function 'fseterr':
  ../../libunistring-0.9.10/lib/fseterr.c:55:6: error: 'FILE' {aka 'struct _iobuf'} has no member named '_flag'
     55 |   fp_->_flag |= _IOERR;
        |      ^~
  make[2]: *** [Makefile:5917: config.h] Error 1
```

https://sourceforge.net/p/mingw-w64/mailman/message/37257960/
> I don't think it is a regression, because there is `MSVC` in the comment on that line, and if MSVC 
was used to compile that file (with UCRT), the same error will occur. It's probably better to add 
`#elif defined _UCRT` above that line, and define a private `struct __crt_stdio_stream_data` there.

I am going to attempt a patch as advised, but I have no idea how to contribute to gnulib, and gnulib source is embedded in many things...

**Comment by jeremyd2019** [comment]
This was broken by commit [fe226dd9028f3f64bef4b4384edb1ef2c95c15f4](https://sourceforge.net/p/mingw-w64/mingw-w64/ci/fe226dd9028f3f64bef4b4384edb1ef2c95c15f4/) to mingw-w64.

**Comment by jeremyd2019** [comment]
https://sourceforge.net/p/mingw-w64/mailman/message/37258177/
> On Thu, 8 Apr 2021, Jeremy Drake via Mingw-w64-public wrote:
> 
> > It looks like gnulib would Do The Right Thing if _IOERR were not defined
> > for UCRT.  Could you do that?
> 
> I've confirmed that the following patch allows gnulib in libunistring to
> build.
> 
> ```diff
> diff --git a/mingw-w64-headers/crt/stdio.h b/mingw-w64-headers/crt/stdio.h
> index c7617798..8f795f62 100644
> --- a/mingw-w64-headers/crt/stdio.h
> +++ b/mingw-w64-headers/crt/stdio.h
> @@ -120,6 +120,7 @@ extern FILE (* __MINGW_IMP_SYMBOL(_iob))[]; /* A pointer to an array of FILE */
>  #define stderr (__acrt_iob_func(2))
>  #endif
> 
> +#ifndef _UCRT
>  #define _IOREAD 0x0001
>  #define _IOWRT 0x0002
> 
> @@ -135,6 +136,7 @@ extern FILE (* __MINGW_IMP_SYMBOL(_iob))[]; /* A pointer to an array of FILE */
>  #ifdef _POSIX_
>  #define _IOAPPEND 0x0200
>  #endif
> +#endif
> 
>  #define _TWO_DIGIT_EXPONENT 0x1
>

**Comment by jeremyd2019** [comment]
This should be fixed upstream mingw-w64 now in commit [1c773877f4a13c8bd7bfb8da80e1e8761a889f51](https://sourceforge.net/p/mingw-w64/mingw-w64/ci/1c773877f4a13c8bd7bfb8da80e1e8761a889f51/).
